jscomp/others/js_dict.mli

```diff
@@ -53,8 +56,25 @@ Array.iter (fun key -> Js.log (Js_dict.unsafeGet dic key)) (Js_dict.keys dict)
 external set : 'a t -> key -> 'a -> unit = "" [@@bs.set_index]  
 (** [set dict key value] sets the [key]/[value] in [dict] *)
 
-external keys : 'a t -> string array = "Object.keys" [@@bs.val]
-(** [keys dict] returns all the keys in the dictionary [dict]*)
+external keys : 'a t -> key array = "Object.keys" [@@bs.val]
+(** [keys dict] returns the keys in [dict] *)
 
-external empty : unit -> 'a t = "" [@@bs.obj]
-(** [empty ()] returns en empty dictionary *)
+(* external entries : 'a t -> (key * 'a) array = "Object.entries" [@@bs.val] *)
+val entries : 'a t -> (key * 'a) array
+(** [entries dict] returns the key value pairs in [dict] (ES2017) *)
+
+(* external values : 'a t -> 'a array = "Object.values" [@@bs.val] *)
+val values : 'a t -> 'a array
+(** [entries dict] returns the values in [dict] (ES2017) *)
+
+val fromList : (key * 'a) list -> 'a t
+(** [fromList entries] creates a new dictionary using with containing each
+[(key, value)] pair in [entries] *)
+
+val fromArray : (key * 'a) array -> 'a t
+(** [fromArray entries] creates a new dictionary using with containing each
+[(key, value)] pair in [entries] *)
+
+val map : ('a -> 'b) -> 'a t -> 'b t
```

Comment:
Obviously it has lots of uses, but the reason it was added now is specifically for the encoding of dictionaries, where you need to encode all the values a dict holds before you encode the dict as a whole. So you can use this to compose the `object_` and, say, `int` encoders just like you would compose the `array` and `int` encoders using `Array.map`.